src/ImageSharp/Formats/Gif/GifEncoderCore.cs

```diff
@@ -82,6 +86,10 @@ public void Encode<TPixel>(Image<TPixel> image, Stream stream)
             Guard.NotNull(image, nameof(image));
             Guard.NotNull(stream, nameof(stream));
 
+            this.gifMetaData = image.MetaData.GetGifMetaData() ?? new GifMetaData();
```

Comment:
@JimBobSquarePants not sure if this was @dlemstra 's original idea. This way we can not share common properties between different formats.

@dlemstra isn't what you meant actually to have entries for individual properties? Eg. `image.MetaData.GetGifColorTableMode()`, `image.Metadata.GlobalColorTableLength` ?

I'm not saying this is the ideal way to go but we should consider pros/contras.

Reply:
I don't mind not sharing. I avoids potential ambiguity and ensures future format support can't break anything.